internal/azure/detect_resource_modification_test.go

```diff
@@ -200,13 +202,28 @@ func TestDetectResourceModification_SpawnedMonitorHasCorrectStates(t *testing.T)
 	result, err := detector.Analyze(log, putInteraction)
 
 	g.Expect(err).ToNot(HaveOccurred())
-	g.Expect(result.Spawn).To(HaveLen(1))
-
-	// Verify the spawned monitor is configured with correct states
-	if m, ok := result.Spawn[0].(*MonitorProvisioningState); ok {
-		g.Expect(m.targetStates).To(ContainElements("Creating", "Updating"))
-		g.Expect(m.baseURL).To(Equal(baseURL))
+	g.Expect(result.Spawn).To(HaveLen(2), "Should spawn two monitors, one for Creating and one for Updating")
+
+	// Verify the spawned monitors are configured correctly
+	var creatingMonitor, updatingMonitor *MonitorProvisioningState
```

Comment:
We dislike type obfuscation like this - be explicit about the type of each variable.

```suggestion
    var creatingMonitor *MonitorProvisioningState
    var updatingMonitor *MonitorProvisioningState
```